apps/aecore/src/aec_pow_test.erl

```diff
@@ -0,0 +1,42 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2017, Aeternity Anstalt
+%%% @doc
+%%%    A library providing PoW generation and verification for testing purposes
+%%%    The solution is always 42.
+%%%
+%%% @end
+%%%-------------------------------------------------------------------
+-module(aec_pow_test).
+
+-behaviour(aec_pow).
+
+-export([generate/3,
+         verify/4]).
+
+-ifdef(TEST).
+-compile([export_all, nowarn_export_all]).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+-include("pow.hrl").
+
+%%%=============================================================================
+%%% API
+%%%=============================================================================
+-spec generate(Data :: aec_hash:hashable(), Target :: aec_pow:sci_int(),
+               Nonce :: aec_pow:nonce()) -> aec_pow:pow_result().
+generate(_Data, _Target, Nonce) ->
```

Comment:
The refactoring is too shallow. A generic PoW framework, should make any PoW pluggable. The Test PoW still respects restrictions native to Cuckoo, which may make no sense in the new Test context.

Reply:
is it ok in terms of interface though? i see the purpose of a generic PoW framework as being generating data and verifying said data, so the thing missing here is a way to generically parameterize the invocation of these methods, am i reading it right?